src/main/java/com/google/maps/DirectionsApi.java

```diff
@@ -47,9 +49,10 @@ public static DirectionsApiRequest getDirections(GeoApiContext context,
     return newRequest(context).origin(origin).destination(destination);
   }
 
-  static class Response implements ApiResponse<DirectionsRoute[]> {
+  static class Response implements ApiResponse<DirectionsResult> {
```

Comment:
Introducing a full results vs an elided just the routes result was a reasonably light change. WDYT?